executor/stale_txn_test.go

```diff
@@ -996,3 +998,96 @@ func (s *testStaleTxnSerialSuite) TestStaleReadPrepare(c *C) {
 	tk.MustExec(fmt.Sprintf(`set transaction read only as of timestamp '%s'`, time1.Format("2006-1-2 15:04:05.000")))
 	c.Assert("execute p1", NotNil)
 }
+
+func (s *testStaleTxnSuite) TestStmtCtxStaleFlag(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	defer tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t (id int)")
+	time.Sleep(2 * time.Second)
+	time1 := time.Now().Format("2006-1-2 15:04:05")
+	testcases := []struct {
+		sql          string
+		hasStaleFlag bool
+	}{
+		// assert select as of statement
+		{
+			sql:          fmt.Sprintf("select * from t as of timestamp '%v'", time1),
```

Comment:
Tests based on `explain format="brief"` will be better. Current tests does not ensure it won't be executed on tiflash. It only asserts the truth of `isStaleness`. `physical_plan_test.go` has some good examples.

But I am OK to the current tests anyway. Not a necessary change.